tests/examples/shims/awesome.lua

```diff
@@ -68,7 +69,11 @@ function awesome.load_image(file)
 
     table.insert(surfaces, s)
 
-    return s._native, not s and "Could not load surface from "..file or nil, s
+    return s._native, not s and "Could not load surface from "..file or nil, s]]
+
+    local image = lgi.cairo.ImageSurface.create_from_png(file)
+
+    return image, nil
```

Comment:
Just do `awesome.load_image = lgi.cairo.ImageSurface.create_from_png`. The extra `nil` value isn't needed (the C code doesn't return it in the success case either).